src/sparse/smmp.rs

```diff
@@ -439,8 +439,7 @@ mod test {
 
         let mut c_data = vec![0.; c_indices.len()];
         let mut tmp = [0.; 5];
-        let mut c = crate::CsMatViewMutI::new_trusted_mut_view(
-            crate::CSR,
+        let mut c = crate::CsMatViewMutI::new(
```

Comment:
I'm really glad this change manages to make this work, I didn't like the call to `new_trusted_mut_view` I had to introduce before, it felt unidiomatic.

Reply:
So now I'm wondering if that means we can remove `new_trusted_mut_view` ?

Reply:
Yes, that worked, Moved `new_trusted` to the appropriate traits restrictions and this worked

Reply:
That was a typo it seems, it should call `new_trusted` so we don't check our known valid matrice. Fixed

Reply:
Good catch, I had not noticed this would check the invariants unnecessarily.